src/imgix-core-js.js

```diff
@@ -124,6 +124,74 @@
       }
     };
 
+    ImgixClient.prototype.buildSrcSet = function (path, params) {
+      var width = params ? params['w'] : undefined;
+      var height = params ? params['h'] : undefined;
+      var aspectRatio = params ? params['ar'] : undefined;
+
+      // determines if an aspect ratio value is in the correct format 'w:h'
+      var isValidFormat = function() {
+        return /^\d+(\.\d+)?:\d+(\.\d+)?$/.test(aspectRatio);
+      }();
+
+      if (aspectRatio && !isValidFormat) {
+        throw new Error('The \'ar\' parameter key must follow the format w:h');
+      }
+
+      var fixedWidth = ((width && height) || (width && aspectRatio) || (height && aspectRatio)) ? true : false;
```

Comment:
This also feels contrived to me. Why not just put the condition in the conditional?

```js
// If two of the three are defined, this is a fixed-dimension srcset
if ((width && height) || (width && aspectRatio) || (height && aspectRatio)) {
  return this._buildDPRSrcSet(path, params);
} else {
  return this._buildSrcSetPairs(path, params);
}
```